pkg/kv/kvserver/logstore/sideload_disk.go

```diff
@@ -279,3 +283,57 @@ func (ss *DiskSideloadStorage) String() string {
 	fmt.Fprintf(&buf, "(%d files)\n", count)
 	return buf.String()
 }
+
+// mkdirAllAndSyncParents creates the given chain of directories, if any
+// subdirectory is missing, and syncs parents of all the created directories.
+// The base directory must already exist. The directories are created using the
+// provided permissions mask, with the same semantics as in os.MkdirAll.
+//
+// For example, if dirs is [x, y, z], then this func creates all the missing
+// subdirectories in base/x/y/z. If base is the lowest previously existing
+// directory then base, x, and y are synced. If x existed too, then only x and y
+// are synced. If z existed before then there are no syncs.
+//
+// NB: path syntax in the examples above may differ, because components of the
+// path are combined using the provided filesystem interface.
+func mkdirAllAndSyncParents(fs vfs.FS, base string, dirs []string, perm os.FileMode) error {
+	paths := append(make([]string, 0, len(dirs)+1), base)
+	for i, part := range dirs {
+		paths = append(paths, fs.PathJoin(paths[i], part))
+	}
+
+	// Find the lowest existing directory in the hierarchy.
+	existing := -1
+	for i := len(paths) - 1; i >= 0; i-- {
+		if _, err := fs.Stat(paths[i]); err == nil {
+			existing = i
+			break
+		} else if !oserror.IsNotExist(err) {
+			return errors.Wrapf(err, "could not get dir info: %s", paths[i])
+		}
+	}
+	if existing == -1 {
+		return errors.Newf("base dir does not exist: %s", base)
+	}
```

Comment:
This can be problematic if someone drops the `auxiliary` directory for fun (for example, they see it's empty and assume it is then safe to remove it).

Maybe we should walk all the way up to the Pebble root directory instead, to be more robust.